autogpt_platform/backend/schema.prisma

```diff
@@ -272,11 +385,220 @@ model UserBlockCredit {
+
   amount Int
   type   UserBlockCreditType
 
-  isActive Boolean @default(true)
-  metadata Json?
+  isActive         Boolean         @default(true)
+  metadata         Json?
+  UserAccounting   UserAccounting? @relation(fields: [userAccountingId], references: [id])
+  userAccountingId String?
 
   @@id(name: "creditTransactionIdentifier", [transactionKey, userId])
 }
+
+// This is the top level accounting table for a user
+model UserAccounting {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  transactions UserBlockCredit[]
+
+  usdBalance Float @default(0.0)
+
+  subscription   UserSubscription? @relation(fields: [subscriptionId], references: [id])
+  subscriptionId String?
+}
+
+// This table tracks a user's subscription to autogpt
+model UserSubscription {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isSubscribed          Boolean   @default(false)
+  subscriptionStartDate DateTime?
+
+  isCancelling        Boolean   @default(false)
+  subscriptionEndDate DateTime?
+
+  subscriptionPlanId String?
+  subscriptionPlan   SubsctiptionPlan? @relation(fields: [subscriptionPlanId], references: [id])
+
+  hasInitiatedChargeBackProcess Boolean          @default(false)
+  chargeBackProcessInitiatedAt  DateTime?
+  UserAccounting                UserAccounting[]
+
+  @@index([isSubscribed, isCancelling])
+}
+
+// This table tracks the different subscription plans available to a user
+model SubsctiptionPlan {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name        String
+  description String
+  planIconUrl String?
+
+  creditsPerMonth Int
+  usdPrice        Float
+
+  isDeleted        Boolean            @default(false)
+  UserSubscription UserSubscription[]
+}
+
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+//////////////   Store TABLES    /////////////////////
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+
+model Profile {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String?
+  user   User?   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isGroupProfile Boolean    @default(false)
+  groupId        String?
+  group          UserGroup? @relation(fields: [groupId], references: [id])
+
+  username    String @unique
+  description String
+
+  website  String?
+  github   String?
+  linkedin String?
+  facebook String?
+  twitter  String?
```

Comment:
x seems to short a name...

Reply:
Maybe it’s better to have just a list of platform links and an enum (if not string and handle in Python) for keys rather than row by row if we want to to add like mastadon or threads or printable or whatever

Reply:
Yeah probably. a better idea, I'll change it to a list of strings.